vernac/declare.mli

```diff
@@ -598,6 +598,8 @@ val admit_obligations : pm:OblState.t -> Names.Id.t option -> OblState.t
 
 val check_program_libraries : unit -> unit
 
+val program_inference_hook : Environ.env -> Evd.evar_map -> Evar.t -> (Evd.evar_map * EConstr.t) option
```

Comment:
Maybe an opportunity to document and give this a better name?

Reply:
I don't know how to call it and I don't know either if it is worth to keep it.